src/core/server/integration_tests/http/lifecycle_handlers.test.ts

```diff
@@ -391,3 +300,45 @@ describe('core lifecyle handers with restrict internal routes enforced', () => {
     });
   });
 });
+
+describe('core lifecycle handlers with no strict client version check', () => {
+  const testRoute = '/version_check/test/route';
+  let server: HttpService;
+  let innerServer: HttpServerSetup['server'];
+  let router: IRouter;
+
+  beforeEach(async () => {
+    const configService = createConfigService({
+      server: {
+        versioned: {
+          strictClientVersionCheck: false,
+          handlerResolution: 'newest',
+        },
+      },
+    });
+    server = createHttpServer({ configService });
+    await server.preboot({ context: contextServiceMock.createPrebootContract() });
+    const serverSetup = await server.setup(setupDeps);
+    router = serverSetup.createRouter('/');
+    router.get({ path: testRoute, validate: false }, (context, req, res) => {
+      return res.ok({ body: 'ok' });
+    });
+    innerServer = serverSetup.server;
+    await server.start();
+  });
+
+  afterEach(async () => {
+    await server.stop();
+  });
+
+  it('accepts requests that do not include a version header', async () => {
+    await supertest(innerServer.listener).get(testRoute).expect(200, 'ok');
+  });
+
+  it('accepts requests with any version passed in the version header', async () => {
+    await supertest(innerServer.listener)
+      .get(testRoute)
+      .set(versionHeader, 'what-have-you')
+      .expect(200, 'ok');
+  });
```

Comment:
This is the most important change/difference. We do not block for version headers.